src/oic/utils/sdb.py

```diff
@@ -1018,44 +884,49 @@ def get_client_id_for_session(self, sid):
 
     def get_client_ids_for_uid(self, uid: str) -> List[str]:
         """Return client_ids for a given uid."""
-        return self._db.get_client_ids_for_uid(uid)
+        return [self.get_client_id_for_session(sid) for sid in self._db.get_by_uid(uid)]
 
-    def get_verify_logout(self, uid: str) -> str:
-        """Return the key for logout verification."""
-        return self._db.get_verified_logout(uid)
+    def get_verify_logout(self, uid: str) -> Dict:
+        """Return the dictionary for logout verification."""
+        res = {}
+        for sid in self._db.get_by_uid(uid):
+            _dict = self._db[sid]
+            try:
+                res[sid] = _dict["verified_logout"]
+            except KeyError:
+                res[sid] = None
+        return res
 
     def set_verify_logout(self, uid: str) -> None:
         """Save the key that is used for logout verification."""
-        logout_key = uuid.uuid4().urn
-        for sid in self.get_sids_from_uid(uid):
-            self.update(sid, "verified_logout", logout_key)
-
-    def get_token_ids(self, uid: str) -> List[str]:
-        """Return id_tokens for given uid."""
-        return self._db.get_token_ids(uid)
+        for sid in self._db.get_by_uid(uid):
+            _dict = self._db[sid]
+            _dict["verified_logout"] = uuid.uuid4().urn
+
+    def get_id_token(self, uid: str) -> Dict:
+        res = {}
+        for sid in self._db.get_by_uid(uid):
+            _dict = self._db[sid]
+            try:
+                res[_dict["client_id"]] = _dict["id_token"]
+            except KeyError:
+                pass
+        return res
 
-    def is_revoke_uid(self, uid: str) -> bool:
-        """Return if the uid session has been revoked."""
-        return self._db.is_revoke_uid(uid)
+    def is_revoke_uid(self, uid: str) -> Dict:
+        res = {}
+        for sid in self._db.get_by_uid(uid):
+            try:
```

Comment:
This can be rewritten using `get()`.

The except part is not tested.

Reply:
And a third provider related issue.

Reply:
The same applies here: If not part of PR, shouldn't be here.

Reply:
OK